src/layouts/components/Editor/Iframe.ts

```diff
@@ -0,0 +1,118 @@
+import { Node, nodeInputRule, nodePasteRule } from "@tiptap/core"
+import { ReactNodeViewRenderer } from "@tiptap/react"
+import _ from "lodash"
+
+import { LegacyBlockView } from "./LegacyBlockView"
+
+export interface IframeOptions {
+  allowFullscreen: boolean
+  HTMLAttributes: {
+    [key: string]: any
+  }
+}
+
+declare module "@tiptap/core" {
+  interface Commands<ReturnType> {
+    iframe: {
+      /**
+       * Add an iframe
+       */
+      setIframe: (options: { src: string }) => ReturnType
+    }
+  }
+}
+
+export default Node.create<IframeOptions>({
+  name: "iframe",
+
+  group: "block",
+  atom: true,
+
+  draggable: true,
+
+  defining: true,
+
+  addOptions() {
+    return {
+      allowFullscreen: true,
+      HTMLAttributes: {
+        class: "iframe-wrapper",
+      },
+    }
+  },
+
+  addAttributes() {
+    return {
+      src: {
+        default: null,
+      },
+      frameborder: {
+        default: 0,
+      },
+      allowfullscreen: {
+        default: this.options.allowFullscreen,
+        parseHTML: () => this.options.allowFullscreen,
+      },
+      width: {
+        default: null,
+      },
+      height: {
+        default: null,
+      },
+      style: {
+        default: null,
+      },
+    }
+  },
+
+  parseHTML() {
+    return [
+      {
+        tag: "iframe",
+      },
+    ]
+  },
+
+  renderHTML({ HTMLAttributes }) {
+    return ["div", this.options.HTMLAttributes, ["iframe", HTMLAttributes]]
+  },
+
+  addCommands() {
+    return {
+      setIframe: (options: { src: string }) => ({ tr, dispatch }) => {
+        const { selection } = tr
+        const node = this.type.create(options)
+
+        if (dispatch) {
+          tr.replaceRangeWith(selection.from, selection.to, node)
+        }
+
+        return true
+      },
+    }
+  },
+
+  addPasteRules() {
+    return [
+      nodePasteRule({
+        find: /<iframe (.*)><\/iframe>/g,
```

Comment:
what happens if the user has an iframe? 
do we input the iframe? do we care cases like `<iframe></iframe>` (this will not be parsed cuz there is no spacing bet the `e` and hte closing anchor tag)

Reply:
we never input anything. we just parse the source text (if going from old -> new) and check if there exists `iframe` tags.

if there are iframe tags per your example, they never did anything in the first place ._.